firmware/projects/Demo/CANErrors/inc/app.h

```diff
@@ -0,0 +1,110 @@
+/// @author Luai Bashar
+/// @date 2024-11-10
+/// @brief Functions and types that will be used in CANErrors main
+
+#pragma once
+
+#include "../generated/can/error_can_messages.h"
+#include "shared/comms/can/can_bus.h"
+
+/***************************************************************
+    App-level objects
+***************************************************************/
+
+// Interface to set/send all possible errors for the system
+class ErrorHandler {
+    private:
+        // Object that holds a 64 bit int, each bit representing an error
+        generated::can::TMS_ERROR error_msg{};
+    public:
+        // Sets the error based on the error index given
+        void setError(uint32_t error) {
+            if (error < 0 || error > 63) {
+                std::cerr << "Error index must be between 0 and 63!" << std::endl;
+                return;
+            }
+
+            error_msg.errors |= (1ULL << error);
+        }
+
+        // Sends the error message through the provided CAN bus
+        void sendError(shared::can::CanBus error_can_bus) {
+            error_can_bus.Send(error_msg);
+
+            // Reset after a send to not send duplicate errors
+            resetError();
+        }
```

Comment:
Made the CAN bus to send the error object to as a parameter for this function. This is okay if the bus is changing and will not always be the same one.

However if it is always the same one, this can just be set as a private class variable to be used internally by this method. Let me know thoughts on what is expected.

Reply:
The error message is only defined for a specific bus (it's only present in one dbc) and doesn't make sense to send on an arbitrary bus. We will likely generate this class so don't worry about that part of it

Reply:
Alright, should I leave it as be or define the bus as a private member then?

Reply:
Just add a comment in the code for now